### PR TITLE
refactor(course-cards-next): use carousel

### PR DESCRIPTION
**Pull request by MichaelSolati**
Changes proposed in this pull request:

- Update course cards partial to use carousel
- Carousel tracks active card when tabbing and using arrow keys.
- Carousel scrolls when using forward/back buttons.

When you're ready to submit your PR, don't forget to add the `$-presubmit` label.

### CODE REVIEW

**Comment by samthor** [comment]
This is fine but the new homepage doesn't actually load the "Home" entrypoint.

I think you want to add this to "homepage-next.njk":

```
pageScripts:
  - '/js/home.js'
```

(which matches "homepage.njk")

Approved but please do this.

**Review by samthor — APPROVED** [review]

**Comment by devnook** [comment]
I think the keyboard interaction is broken, e.g:

1. Hit tab to focus the "<" button
2. Hit enter to activate it
3. Carousel scrolls right (>) and 2nd items gets focus.

or:

1. Hit tab to focus the "<" button
2. Hit tab to enter into carousel. 1st item gets focused
3. Hit right arrow. 2nd item gets focused
4. Hit right arrow again. Nothing happens.
5. You can still go left and right between first and second item, but cannot get to 3rd and 4th.

File: src/site/_includes/partials/course-cards-next.njk [hunk 1]
```diff
@@ -1,27 +1,37 @@
...
-      </a>
-    {% endif %}
-  {% endfor %}
-</nav>
+<web-carousel>
+  <div class="carousel">
+    <button class="icon-button" data-direction="prev" aria-label="back">
+      {% include "icons/carat-back.svg" %}
+    </button>
+    <div class="carousel__track reel" data-scroll="snap" aria-label="available courses">
```
Comment (devnook) [inline src/site/_includes/partials/course-cards-next.njk:8]: please add back the scrollbar class to that scrollbar is visible when idle.

File: src/lib/components/Carousel/index.js [hunk 2]
```diff
@@ -60,90 +49,98 @@ class Carousel extends BaseElement {
...
     this._items.forEach((e, i) =>
-      e.addEventListener('focusin', () => this.moveSlide(i - this._index)),
+      e.addEventListener('focusin', () => this._moveSlide(i - this._index)),
     );
+    this._carouselTrack.addEventListener('keyup', this._onKeyup);
 
-    this._carouselTrack.addEventListener('scroll', this._onScroll);
-
-    this.moveSlide(0, false);
+    this._moveSlide(0);
```
Comment (devnook) [inline src/lib/components/Carousel/index.js:63]: Why this? It gives the focus on first card after render, even if no user action indicated they indicated to focus that card. 

It gives an additional side effect, that if they load the page, dont even scroll to the carousel, and hit accidentally enter, they will navigate to a random page.

**Review by devnook — CHANGES_REQUESTED** [review]
Apart from the comment left, lgtm. I would love to see a test for this though at some point.

File: src/lib/components/Carousel/index.js (hunk 2 above)
Comment (MichaelSolati) [reply src/lib/components/Carousel/index.js:63]: This is a total remnant of the way it used to work, it definitely shouldn't be here now.

**Comment by MichaelSolati** [comment]
@devnook PTAL

**Review by devnook — APPROVED** [review]

**Pull request merged**